overlord/snapstate/autorefresh.go

```diff
@@ -172,20 +173,35 @@ func (m *autoRefresh) refreshScheduleWithDefaultsFallback() (ts []*timeutil.Sche
 		return nil, "managed", nil
 	}
 
-	refreshScheduleStr := defaultRefreshSchedule
 	tr := config.NewTransaction(m.state)
-	err = tr.Get("core", "refresh.schedule", &refreshScheduleStr)
+	err = tr.Get("core", "refresh.timer", &scheduleAsStr)
 	if err != nil && !config.IsNoOption(err) {
 		return nil, "", err
 	}
 
-	refreshSchedule, err := timeutil.ParseLegacySchedule(refreshScheduleStr)
+	if scheduleAsStr == "" {
+		logger.Noticef("refresh.timer is not set, fallback to legacy refresh.schedule")
+		// try legacy refresh schedule config option
+		err = tr.Get("core", "refresh.schedule", &scheduleAsStr)
+		if err != nil && !config.IsNoOption(err) {
+			return nil, "", err
+		}
+		ts, err = timeutil.ParseLegacySchedule(scheduleAsStr)
+		if err != nil {
+			logger.Noticef("cannot use refresh.schedule configuration: %s", err)
+		}
+	} else {
+		ts, err = timeutil.ParseSchedule(scheduleAsStr)
+		if err != nil {
+			logger.Noticef("cannot use refresh.timer configuration: %s", err)
+		}
+	}
+
 	if err != nil {
-		logger.Noticef("cannot use refresh.schedule configuration: %s", err)
```

Comment:
AFAICT what will happen is that if there is no option set at all (fresh state) we now set the refresh.timer to the default value. This ok but it has the disadvantage that if we ever change the default we cannot easily tell if the user set the option or if we set the option. I think I would slightly prefer if we just keep the option unset in the state and use the (in-code) default string. 

Maybe the old code was already doing the above (which is slightly sad). But I hope with the new code we can do better.

Reply:
+1!

Let's please make sure the option remains empty when it was empty.